RecoTracker/MkFitCore/interface/Config.h

```diff
@@ -0,0 +1,433 @@
+#ifndef RecoTracker_MkFitCore_interface_Config_h
+#define RecoTracker_MkFitCore_interface_Config_h
+
+#include <algorithm>
+#include <cmath>
+#include <string>  // won't compile on clang gcc for mac OS w/o this!
+#include <map>
+#include <vector>
+
+namespace mkfit {
+
+  // Cram this in here for now ...
+  class TrackerInfo;
+  class IterationsInfo;
+  //------------------------------------------------------------------------------
+
+  enum PropagationFlagsEnum {
+    PF_none = 0,
+
+    PF_use_param_b_field = 0x1,
+    PF_apply_material = 0x2
+  };
+
+  struct PropagationFlags {
+    union {
+      struct {
+        bool use_param_b_field : 1;
+        bool apply_material : 1;
+        // Could add: bool use_trig_approx  -- now Config::useTrigApprox = true
+        // Could add: int  n_iter : 8       -- now Config::Niter = 5
```

Comment:
this flexibility could still be useful.
I created an issue trackreco/cmssw#77

It still seems more practical to keep this comment here:
- the placement of the variables is clear
- in case some other extension of the flag comes in, this comment would hold/reserve the space for that potential use case